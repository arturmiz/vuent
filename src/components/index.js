import './Vnt.vue';
export { default as VntAutosuggest } from './autosuggest';
export { default as VntButton } from './button';
export { default as VntCheckbox } from './checkbox';
export { default as VntDialog } from './dialog';
export { default as VntHeader } from './header';
export { default as VntIcon } from './icon';
export { default as VntInput } from './input';
export { default as VntLink } from './link';
export { default as VntListView } from './listview';
export { default as VntNavView } from './navview';
export { default as VntPivot } from './pivot';
export { default as VntRadio } from './radio';
export { default as VntRating } from './rating';
export { default as VntSelect } from './select';
export { default as VntSlider } from './slider';
export { default as VntToggle } from './toggle';
