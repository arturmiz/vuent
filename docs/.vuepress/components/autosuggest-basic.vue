<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestBasic',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
