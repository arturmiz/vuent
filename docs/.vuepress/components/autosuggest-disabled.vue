<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries" disabled></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestDisabled',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
