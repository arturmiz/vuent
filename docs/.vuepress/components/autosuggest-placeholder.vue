<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"
                     placeholder="Type something..."></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestPlaceholder',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
