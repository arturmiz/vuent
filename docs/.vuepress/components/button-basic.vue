<template>
  <div class="playground">
    <vnt-button>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonBasic',
  components: {
    VntButton
  }
}
</script>
