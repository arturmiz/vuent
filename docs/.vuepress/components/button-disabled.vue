<template>
  <div class="playground">
    <vnt-button disabled>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonDisabled',
  components: {
    VntButton
  }
}
</script>
