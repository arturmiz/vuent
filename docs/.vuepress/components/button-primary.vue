<template>
  <div class="playground">
    <vnt-button primary>Primary</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonPrimary',
  components: {
    VntButton
  }
}
</script>
