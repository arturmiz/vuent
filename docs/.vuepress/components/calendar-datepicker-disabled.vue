<template>
  <div class="playground">
    <vnt-calendar-datepicker v-model="date" disabled />
    <div class="result">date: {{ date }}</div>
  </div>
</template>

<script>
import { VntCalendarDatepicker } from "../../../src/components";

export default {
  name: "CalendarDatepickerBasic",
  components: {
    VntCalendarDatepicker
  },
  data() {
    return {
      date: "2020-04-05"
    };
  }
};
</script>
