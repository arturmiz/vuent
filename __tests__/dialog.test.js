import { createLocalVue, mount } from '@vue/test-utils';
import { VntDialog } from '@/components';
import { isInstalled } from './utils';

describe('Dialog', () => {
  let wrapper;

  function emittedResult(wrapper, i = 0) {
    const { result } = wrapper.emitted();
    const [eventValue] = result[i];
    return eventValue;
  }

  function openedResult(wrapper) {
    const [[eventValue]] = wrapper.emitted()['update:opened'];
    return eventValue;
  }

  test('can be installed separately', () => {
    const localVue = createLocalVue();
    localVue.use(VntDialog);

    expect(isInstalled(localVue, VntDialog)).toBe(true);
  });

  describe('by default', () => {

    beforeAll(() => {
      wrapper = mount(VntDialog);
    });

    test('title is empty', () => {
      expect(wrapper.vm.title).toBe('');
    });

    test('content is empty', () => {
      expect(wrapper.vm.content).toBe('');
    });

    test('dialog is hidden', () => {
      expect(wrapper.vm.opened).toBe(false);
    });

    test('nothing is rendered as dialog should be hidden', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('by default, when opened', () => {

    beforeAll(() => {
      wrapper = mount(VntDialog, {
        propsData: {
          opened: true
        }
      });
    });

    test('closes on ESC key', () => {
      wrapper.find({ ref: 'dialog' }).trigger('keyup.esc');

      expect(openedResult(wrapper)).toBe(false);
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe('default dismiss button clicked', () => {

      beforeAll(() => {
        wrapper.find({ ref: 'vnt-dialog-btn-dismiss' }).trigger('click');
      });

      test('emits correct result', () => {
        expect(emittedResult(wrapper)).toBe(null);
      });

      test('closes dialog', () => {
        expect(openedResult(wrapper)).toBe(false);
      });

    });

  });

  describe('when opened after initialization', () => {

    beforeAll(() => {
      wrapper.setProps({ opened: true });
    });

    describe('on ESC key pressed', () => {

      test('emits correct result', () => {
        expect(emittedResult(wrapper)).toBe(null);
      });

      test('closes dialog', (done) => {
        wrapper.vm.$nextTick(() => {
          wrapper.find({ ref: 'dialog' }).trigger('keyup.esc');

          expect(openedResult(wrapper)).toBe(false);
          done();
        });
      });

    });

  });

  describe('with title and content given', () => {

    beforeAll(() => {
      wrapper = mount(VntDialog, {
        propsData: {
          opened: true,
          title: 'Warning',
          content: 'This operation is not allowed.'
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });

  describe('when custom dismiss button given', () => {

    beforeAll(() => {
      wrapper = mount(VntDialog, {
        propsData: {
          opened: true,
          title: 'Warning',
          content: 'This operation is not allowed.'
        },
        slots: {
          default: '<vnt-dialog-btn-dismiss result="cancel">Close</vnt-dialog-btn-dismiss>'
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe('dismiss button clicked', () => {

      beforeAll(() => {
        wrapper.find({ ref: 'vnt-dialog-btn-dismiss' }).trigger('click');
      });

      test('emits correct result', () => {
        expect(emittedResult(wrapper)).toBe('cancel');
      });

      test('closes dialog', () => {
        expect(openedResult(wrapper)).toBe(false);
      });

    });

  });

  describe('when custom primary button given', () => {

    beforeAll(() => {
      wrapper = mount(VntDialog, {
        propsData: {
          opened: true,
          title: 'Confirm Delete Action',
          content: 'Are you sure you want to delete this item?'
        },
        slots: {
          default: '<vnt-dialog-btn-primary result="ok">Delete</vnt-dialog-btn-primary>'
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe('primary button clicked', () => {

      beforeAll(() => {
        wrapper.find({ ref: 'vnt-dialog-btn-primary' }).trigger('click');
      });

      test('emits correct result', () => {
        expect(emittedResult(wrapper)).toBe('ok');
      });

      test('closes dialog', () => {
        expect(openedResult(wrapper)).toBe(false);
      });

    });

    describe('dismiss button clicked', () => {

      beforeAll(() => {
        wrapper.find({ ref: 'vnt-dialog-btn-dismiss' }).trigger('click');
      });

      test('emits correct result', () => {
        expect(emittedResult(wrapper, 1)).toBe(null);
      });

      test('closes dialog', () => {
        expect(openedResult(wrapper)).toBe(false);
      });

    });

  });

  describe('when custom secondary button given', () => {

    beforeAll(() => {
      wrapper = mount(VntDialog, {
        propsData: {
          opened: true,
          title: 'Confirm Delete Action',
          content: 'Are you sure you want to delete this item?'
        },
        slots: {
          default: '<vnt-dialog-btn-secondary result="skip">Ask later</vnt-dialog-btn-secondary>'
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe('secondary button clicked', () => {

      beforeAll(() => {
        wrapper.find({ ref: 'vnt-dialog-btn-secondary' }).trigger('click');
      });

      test('emits correct result', () => {
        expect(emittedResult(wrapper)).toBe('skip');
      });

      test('closes dialog', () => {
        expect(openedResult(wrapper)).toBe(false);
      });

    });

    describe('dismiss button clicked', () => {

      beforeAll(() => {
        wrapper.find({ ref: 'vnt-dialog-btn-dismiss' }).trigger('click');
      });

      test('emits correct result', () => {
        expect(emittedResult(wrapper, 1)).toBe(null);
      });

      test('closes dialog', () => {
        expect(openedResult(wrapper)).toBe(false);
      });

    });

  });

  describe('when custom primary and secondary buttons given', () => {

    describe('in rendering order', () => {

      beforeAll(() => {
        wrapper = mount(VntDialog, {
          propsData: {
            opened: true,
            title: 'Confirm Delete Action',
            content: 'Are you sure you want to delete this item?'
          },
          slots: {
            default: `
              <vnt-dialog-btn-primary result="ok">Delete</vnt-dialog-btn-primary>
              <vnt-dialog-btn-secondary result="skip">Ask later</vnt-dialog-btn-secondary>
            `
          }
        });
      });

      test('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });

      describe('dismiss button clicked', () => {

        beforeAll(() => {
          wrapper.find({ ref: 'vnt-dialog-btn-dismiss' }).trigger('click');
        });

        test('emits correct result', () => {
          expect(emittedResult(wrapper)).toBe(null);
        });

        test('closes dialog', () => {
          expect(openedResult(wrapper)).toBe(false);
        });

      });

    });

    describe('in mixed order', () => {

      beforeAll(() => {
        wrapper = mount(VntDialog, {
          propsData: {
            opened: true,
            title: 'Confirm Delete Action',
            content: 'Are you sure you want to delete this item?'
          },
          slots: {
            default: `
              <vnt-dialog-btn-secondary result="skip">Ask later</vnt-dialog-btn-secondary>
              <vnt-dialog-btn-primary result="ok">Delete</vnt-dialog-btn-primary>
            `
          }
        });
      });

      test('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });

    });

  });

  describe('when custom primary, secondary and dismiss buttons given', () => {

    beforeAll(() => {
      wrapper = mount(VntDialog, {
        propsData: {
          opened: true,
          title: 'Confirm Delete Action',
          content: 'Are you sure you want to delete this item?'
        },
        slots: {
          default: `
            <vnt-dialog-btn-primary result="ok">Delete</vnt-dialog-btn-primary>
            <vnt-dialog-btn-secondary result="skip">Ask later</vnt-dialog-btn-secondary>
            <vnt-dialog-btn-dismiss result="cancel">Close</vnt-dialog-btn-dismiss>
          `
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe('primary button clicked', () => {

      beforeAll(() => {
        wrapper.find({ ref: 'vnt-dialog-btn-primary' }).trigger('click');
      });

      test('emits correct result', () => {
        expect(emittedResult(wrapper)).toBe('ok');
      });

      test('closes dialog', () => {
        expect(openedResult(wrapper)).toBe(false);
      });

    });

    describe('secondary button clicked', () => {

      beforeAll(() => {
        wrapper.find({ ref: 'vnt-dialog-btn-secondary' }).trigger('click');
      });

      test('emits correct result', () => {
        expect(emittedResult(wrapper, 1)).toBe('skip');
      });

      test('closes dialog', () => {
        expect(openedResult(wrapper)).toBe(false);
      });

    });

    describe('dismiss button clicked', () => {

      beforeAll(() => {
        wrapper.find({ ref: 'vnt-dialog-btn-dismiss' }).trigger('click');
      });

      test('emits correct result', () => {
        expect(emittedResult(wrapper, 2)).toBe('cancel');
      });

      test('closes dialog', () => {
        expect(openedResult(wrapper)).toBe(false);
      });

    });

  });

});
