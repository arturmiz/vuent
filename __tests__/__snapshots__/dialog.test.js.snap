// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog by default nothing is rendered as dialog should be hidden 1`] = `undefined`;

exports[`Dialog by default, when opened renders correctly 1`] = `
<div class="vnt-dialog-overlay">
  <div role="dialog" tabindex="0" aria-label="" class="vnt-dialog">
    <h1 class="vnt-dialog__title"></h1>
    <p class="vnt-dialog__content"></p>
    <div class="vnt-dialog__actions vnt-dialog__actions--single">
      <button type="button" class="vnt-button">
        Cancel
      </button>
    </div>
  </div>
</div>
`;

exports[`Dialog when custom dismiss button given renders correctly 1`] = `
<div class="vnt-dialog-overlay">
  <div role="dialog" tabindex="0" aria-label="Warning" class="vnt-dialog">
    <h1 class="vnt-dialog__title">Warning</h1>
    <p class="vnt-dialog__content">This operation is not allowed.</p>
    <div class="vnt-dialog__actions vnt-dialog__actions--single">
      <button type="button" class="vnt-button">
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`Dialog when custom primary and secondary buttons given in mixed order renders correctly 1`] = `
<div class="vnt-dialog-overlay">
  <div role="dialog" tabindex="0" aria-label="Confirm Delete Action" class="vnt-dialog">
    <h1 class="vnt-dialog__title">Confirm Delete Action</h1>
    <p class="vnt-dialog__content">Are you sure you want to delete this item?</p>
    <div class="vnt-dialog__actions vnt-dialog__actions--three">
      <button type="button" class="vnt-button">
        Delete
      </button>
      <button type="button" class="vnt-button">
        Ask later
      </button>
      <button type="button" class="vnt-button">
        Cancel
      </button>
    </div>
  </div>
</div>
`;

exports[`Dialog when custom primary and secondary buttons given in rendering order renders correctly 1`] = `
<div class="vnt-dialog-overlay">
  <div role="dialog" tabindex="0" aria-label="Confirm Delete Action" class="vnt-dialog">
    <h1 class="vnt-dialog__title">Confirm Delete Action</h1>
    <p class="vnt-dialog__content">Are you sure you want to delete this item?</p>
    <div class="vnt-dialog__actions vnt-dialog__actions--three">
      <button type="button" class="vnt-button">
        Delete
      </button>
      <button type="button" class="vnt-button">
        Ask later
      </button>
      <button type="button" class="vnt-button">
        Cancel
      </button>
    </div>
  </div>
</div>
`;

exports[`Dialog when custom primary button given renders correctly 1`] = `
<div class="vnt-dialog-overlay">
  <div role="dialog" tabindex="0" aria-label="Confirm Delete Action" class="vnt-dialog">
    <h1 class="vnt-dialog__title">Confirm Delete Action</h1>
    <p class="vnt-dialog__content">Are you sure you want to delete this item?</p>
    <div class="vnt-dialog__actions">
      <button type="button" class="vnt-button">
        Delete
      </button>
      <button type="button" class="vnt-button">
        Cancel
      </button>
    </div>
  </div>
</div>
`;

exports[`Dialog when custom primary, secondary and dismiss buttons given renders correctly 1`] = `
<div class="vnt-dialog-overlay">
  <div role="dialog" tabindex="0" aria-label="Confirm Delete Action" class="vnt-dialog">
    <h1 class="vnt-dialog__title">Confirm Delete Action</h1>
    <p class="vnt-dialog__content">Are you sure you want to delete this item?</p>
    <div class="vnt-dialog__actions vnt-dialog__actions--three">
      <button type="button" class="vnt-button">
        Delete
      </button>
      <button type="button" class="vnt-button">
        Ask later
      </button>
      <button type="button" class="vnt-button">
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`Dialog when custom secondary button given renders correctly 1`] = `
<div class="vnt-dialog-overlay">
  <div role="dialog" tabindex="0" aria-label="Confirm Delete Action" class="vnt-dialog">
    <h1 class="vnt-dialog__title">Confirm Delete Action</h1>
    <p class="vnt-dialog__content">Are you sure you want to delete this item?</p>
    <div class="vnt-dialog__actions">
      <button type="button" class="vnt-button">
        Ask later
      </button>
      <button type="button" class="vnt-button">
        Cancel
      </button>
    </div>
  </div>
</div>
`;

exports[`Dialog with title and content given renders correctly 1`] = `
<div class="vnt-dialog-overlay">
  <div role="dialog" tabindex="0" aria-label="Warning" class="vnt-dialog">
    <h1 class="vnt-dialog__title">Warning</h1>
    <p class="vnt-dialog__content">This operation is not allowed.</p>
    <div class="vnt-dialog__actions vnt-dialog__actions--single">
      <button type="button" class="vnt-button">
        Cancel
      </button>
    </div>
  </div>
</div>
`;
